### PR TITLE
Added Mac OSX support and started adding custom types.

### PR DESCRIPTION
**Pull request by aldmbmtl**
1. List Type (still in development)

### CODE REVIEW

**Comment by melMass** [comment]
You should restructure the PR there are too much small breaking edits:
Why are you replacing single quotes by doubles? It's usually better to avoid this kind of reformating if it's not the sole purpose of the PR.
I'll comment the PR itself

File: PyFlow/Core/AGraphCommon.py [hunk 1]
```diff
@@ -16,19 +16,18 @@
 from enum import IntEnum
 import Enums
 
-
+## determines int maximum value
+INT_RANGE_MAX = 99999
```
Comment (melMass) [inline PyFlow/Core/AGraphCommon.py]: Why are you not using `sys.maxint`? If you use python3 there is no maxint anymore but here is a version that works for both (py 2 &3):

~~~python
import struct
maxint = 2 ** (struct.Struct('i').size * 8 - 1) - 1
~~~

**Review by melMass — CHANGES_REQUESTED** [review]
Most of the typo fixes are right but please remove the formatting changes

File: PyFlow/Core/AGraphCommon.py (hunk 1 above)
Comment (IlgarLunin) [reply PyFlow/Core/AGraphCommon.py]: nice. will add this later

**Comment by IlgarLunin** [comment]
@aldmbmtl most of this changes will be inside user packages (Plugins). See #11

File: PyFlow/Core/AGraphCommon.py (hunk 1 above)
Comment (aldmbmtl) [reply PyFlow/Core/AGraphCommon.py]: Does this work cross platform? I know maxint is no longer a thing in py3 but the maxint was breaking on OSX as well as linux even in py2. So I just hardcoded that number in to see if that was the issue and then see if you guys had a suggestion for how to handle that issue. Thanks!

**Pull request closed without merging**